pandas/tseries/tests/test_tslib.py

```diff
@@ -180,6 +180,63 @@ def test_constructor_invalid(self):
         with tm.assertRaisesRegexp(ValueError, 'Cannot convert Period'):
             Timestamp(Period('1000-01-01'))
 
+    def test_constructor_positional(self):
+        # GH 10758
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1)
```

Comment:
can you use ``assertRaisesRegexp`` here to check for things

```
 In [1]: datetime.datetime(2000,1)
TypeError: Required argument 'day' (pos 3) not found
```

doesn't have to be an exact checking, but need to make sure that it is raising the ``datetime.datetime`` constructor error message and not something else.